hmcts/pre-api
# S28 3151: Asset Playback (AMS Implementation)

### JIRA ticket(s)
- S28-3151


### Change description
- Add AMS implementation for `/media-service/vod` endpoint





> [!IMPORTANT]
> This PR requires manual testing by QA. Please notify the QA team @hmcts/pre-rec-evidence-qa.

**QA instructions**
- Look on AMS, see there is NOT a content key policy with a name matching your user id. Call endpoint, see one is created.
- Call endpoint for valid asset, on asset you are trying to watch, see a new locator with a name that matches the user's id.
- See start time is now, end time is one hours time. Call the endpoint again after one minute. See times update.
  -  _NOTE: The time displayed on AMS is UTC so times will be -1 hour of actual time_
- Check response works for viewing the asset (see steps below)
- Call endpoint for a recording that the user does not have access to
- Call endpoint for a recording that does not exist
- Call endpoint for a recording that has been marked as deleted
- On AMS, turn off default endpoint. See when endpoint called, the streaming endpoint is turned back on.

**How to test response works**
1. Go to this website: https://ampdemo.azureedge.net/ 
2. Set URL to either DASH or HLS URL from response
3. Turn on advanced settings
4. Turn on AES Protected
5. In AES token field enter: `Bearer <token>` where `<token>` is the token from the endpoint response
6. Click Update Player
7. See video loads and is playable
